### PR TITLE
Refactor wpt-results.html to use search API for data

### PR DESCRIPTION
**Pull request by mdittmer**
This PR depends on #414.

### CODE REVIEW

**Comment by mdittmer** [comment]
@Hexcles I think I fixed it. PTAL.

File: webapp/components/wpt-results.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 -->
 
 <link rel="import" href="../bower_components/polymer/polymer-element.html">
-<link rel="import" href="../bower_components/polymer/lib/elements/dom-if.html">
+<link rel="imporxzt" href="../bower_components/polymer/lib/elements/dom-if.html">
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: typo?

File: webapp/components/wpt-results.html [hunk 2]
```diff
@@ -354,9 +352,27 @@ <h3>
...
+        if (search) {
+          url += `q=${window.encodeURIComponent(search)}`;
+        }
+        return url;
+      }
+
       ready() {
         super.ready();
         this.loadRuns().then(async runs => {
+          // Add a diff testRun spoof, if needed.
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: Not any more :)

File: webapp/components/wpt-results.html [hunk 3]
```diff
@@ -281,10 +275,9 @@ <h3>
           },
           displayedTests: {
             type: Array,
-            value: []
-          },
-          // Users request to show a diff column.
-          diff: Boolean,
+            computed: 'computeDisplayedTests(searchResults)',
+          },// Users request to show a diff column.
+          diff: Boolean,	          diff: Boolean,
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: ?

File: webapp/components/wpt-results.html [hunk 4]
```diff
@@ -318,6 +315,7 @@ <h3>
         return diff && testRuns && testRuns.length !== 2;
       }
 
+
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: Redundant empty lines.

File: api/query/types_test.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
+// +build small medium
```
Comment (Hexcles) [inline api/query/types_test.go]: This file is not really a test. And the presence of two tags is more like a hack to include this file when building both small and medium tests.

A more common paradigm seen in Go standard library for this kind of testing utilities is to create a new subpackage, e.g. `api/query/querytest` and to import the package in the test cases.

File: webapp/components/wpt-results.html [hunk 6]
```diff
@@ -281,9 +275,8 @@ <h3>
           },
           displayedTests: {
             type: Array,
-            value: []
-          },
-          // Users request to show a diff column.
+            computed: 'computeDisplayedTests(searchResults)',
+          },// Users request to show a diff column.
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: The comment should go to a new line.

File: webapp/components/wpt-results.html [hunk 7]
```diff
@@ -446,59 +443,48 @@ <h3>
       }
 
       refreshDisplayedNodes() {
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: This method generally looks good to me. However, I'm not very confident. If you know how to write web component tests, this is a something we definitely want to test. There's an existing fixture at `webapp/test/wpt-results.html`.

File: webapp/components/wpt-results.html (hunk 6 above)
Comment (mdittmer) [reply webapp/components/wpt-results.html]: Just fixed this locally. Letting https://travis-ci.org/web-platform-tests/wpt.fyi/builds/419660483 follow through before pushing.

**Comment by mdittmer** [comment]
Filed https://github.com/web-platform-tests/wpt.fyi/issues/470. IMHO, testing `refreshDisplayedNodes()` falls under that larger unit of work.

**Comment by Hexcles** [comment]
This is the culprit of #477 . The staging instance of the PR merged right before this one works: https://github.com/web-platform-tests/wpt.fyi/pull/436#issuecomment-413014518

File: webapp/components/wpt-results.html [hunk 8]
```diff
@@ -374,36 +393,18 @@ <h3>
         };
       }
 
-      async fetchResults(testRuns) {
-        if (!testRuns) {
-          return;
+      // fetchResults should be triggered by changes in searchURL or diffRun,
+      // but diffRun is only used in subroutine: refreshDisplayedNodes.
```
Comment (Hexcles) [inline webapp/components/wpt-results.html]: I don't see where `diffRun` is loaded now.

Comment (mdittmer) [reply webapp/components/wpt-results.html]: `fetchDiff` is invoked here: https://github.com/web-platform-tests/wpt.fyi/pull/431/files/343ee0a740cb4a7c46a28e262f076b6584a886cd#diff-ff3585b170ed88df8ef0c2fa8bcc61ecR383

It sets `diffRun` here https://github.com/web-platform-tests/wpt.fyi/pull/431/files/343ee0a740cb4a7c46a28e262f076b6584a886cd#diff-ff3585b170ed88df8ef0c2fa8bcc61ecR415 which is observed here https://github.com/web-platform-tests/wpt.fyi/pull/431/files/343ee0a740cb4a7c46a28e262f076b6584a886cd#diff-ff3585b170ed88df8ef0c2fa8bcc61ecR313

**Pull request merged**